dspace-api/src/test/java/org/dspace/discovery/DiscoveryIT.java

```diff
@@ -0,0 +1,78 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.discovery;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+
+import org.dspace.AbstractIntegrationTestWithDatabase;
+import org.dspace.builder.CollectionBuilder;
+import org.dspace.builder.CommunityBuilder;
+import org.dspace.builder.WorkspaceItemBuilder;
+import org.dspace.content.Collection;
+import org.dspace.content.Community;
+import org.dspace.content.WorkspaceItem;
+import org.dspace.content.factory.ContentServiceFactory;
+import org.dspace.content.service.WorkspaceItemService;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * This class will aim to test Discovery related use cases
+ */
+public class DiscoveryIT extends AbstractIntegrationTestWithDatabase {
+
+    protected WorkspaceItemService workspaceItemService = ContentServiceFactory.getInstance().getWorkspaceItemService();
+    protected SearchService searchService = SearchUtils.getSearchService();
+
+
+    Community community;
+    Collection col;
+
+    WorkspaceItem leftIs;
+    WorkspaceItem rightIs;
+
+    /**
+     * This method will be run before every test as per @Before. It will
+     * initialize resources required for the tests.
+     *
+     * Other methods can be annotated with @Before here or in subclasses
+     * but no execution order is guaranteed
+     */
+    @Before
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        context.turnOffAuthorisationSystem();
+        community = CommunityBuilder.createCommunity(context).withName("Parent Community").build();
+
+        col = CollectionBuilder.createCollection(context, community).build();
+
+        leftIs = WorkspaceItemBuilder.createWorkspaceItem(context, col).withAbstract("headache").build();
+        rightIs = WorkspaceItemBuilder.createWorkspaceItem(context, col).withAbstract("headache").build();
+
+        context.restoreAuthSystemState();
+    }
+
+    @Test
+    public void deleteWorkspaceItemSolrRecordAfterDeletionFromDbTest() throws Exception {
+        context.turnOffAuthorisationSystem();
```

Comment:
Tiny change.  We should *only* be turning off/on authorization around the `deleteAll()` call in this test.  Otherwise, the `search()` calls in this test are running with a Context that has authorization turned off...and that *can* impact the behavior of those searches.

So, please move this & the `restoreAuthSystemState()` call to be surrounding the `deleteAll()` call.

Reply:
Done.